src/wagtail_ai/ai/base.py

```diff
@@ -0,0 +1,57 @@
+from abc import ABCMeta, abstractmethod
+from collections.abc import Mapping
+from typing import Any, Generic, TypedDict, TypeVar
+
+from ..text_splitters.base import BaseTextSplitterLength
+from ..types import AIResponse, TextSplitterProtocol
+
+
+class ChatModelConfig(TypedDict):
+    id: str
+    token_limit: int
+    extra: Mapping[str, Any] | None
+    text_splitter_class: type[TextSplitterProtocol]
+    text_splitter_length_class: type[BaseTextSplitterLength]
+
+
+AIBackendConfig = TypeVar("AIBackendConfig")
+CMC = TypeVar("CMC", bound=ChatModelConfig)
+
+
+class AIBackend(Generic[AIBackendConfig, CMC], metaclass=ABCMeta):
+    text_splitter_class: type[TextSplitterProtocol]
+    chat_model_config: CMC
+
+    def __init__(
+        self, *, config: AIBackendConfig | None, chat_model_config: CMC
```

Comment:
**question:** Do we need two config objects passed in here? Could a `ChatModelConfig` be accessed through the `AIBackendConfig` in cases where the backend needs it?

Reply:
I've put it all in the AIBackendConfig to keep it simple for now.